templates/boltzmann/middleware/honeycomb.ts

```diff
@@ -9,77 +19,179 @@ import beeline from 'honeycomb-beeline'
 import isDev from 'are-we-dev'
 void `{% endif %}`;
 
+let otelTracer: OtelTracer | null = null;
+
+const OTEL_NAMESPACE = 'boltzmann'
+const OTEL_REQ_NAMESPACE = `${OTEL_NAMESPACE}.request`
+const OTEL_REQ_QUERY = `${OTEL_REQ_NAMESPACE}.query`
+const OTEL_REQ_PARAM_NAMESPACE = `${OTEL_REQ_NAMESPACE}.param`
+
+function getOtelTracer() {
+  if (!otelTracer) {
+    otelTracer = otelTrace.getTracer('boltzmann', '1.0.0')
+  }
+  return otelTracer
+}
+
 function trace ({
   headerSources = ['x-honeycomb-trace', 'x-request-id'],
 } = {}) {
-  if (!process.env.HONEYCOMB_WRITEKEY) {
+  if (!isHoneycomb(process.env)) {
     return (next: Handler) => (context: Context) => next(context)
   }
 
-  const schema = require('honeycomb-beeline/lib/schema')
-  const tracker = require('honeycomb-beeline/lib/async_tracker')
-
-  return function honeycombTrace (next: Handler) {
-    return (context: Context) => {
-      const traceContext = _getTraceContext(context)
-      const trace = beeline.startTrace({
-        [schema.EVENT_TYPE]: 'boltzmann',
-        [schema.PACKAGE_VERSION]: '1.0.0',
-        [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
-        [schema.TRACE_ID_SOURCE]: traceContext.source,
-        'request.host': context.host,
-        'request.original_url': context.url.href,
-        'request.remote_addr': context.remote,
-        'request.method': context.method,
-        'request.scheme': context.url.protocol,
-        'request.path': context.url.pathname,
-        'request.query': context.url.search
-      },
-      traceContext.traceId,
-      traceContext.parentSpanId,
-      traceContext.dataset)
-
-      if (isDev()) {
-        context._honeycombTrace = trace
-      }
-
-      if (traceContext.customContext) {
-        beeline.addContext(traceContext.customContext)
-      }
-
-      if (!trace) {
-        return next(context)
-      }
-
-      const boundFinisher = beeline.bindFunctionToTrace((response: ServerResponse) => {
-        beeline.addContext({
-          'response.status_code': String(response.statusCode)
+  if (isOtel(process.env)) {
+    return function honeycombOtelTrace (next: Handler) {
+      return async (context: Context) => {
+        const tracer = getOtelTracer()
+        let carrier = {}
+
+        // Start a parent span
+        const parentSpan = tracer.startSpan(`${context.method} ${context.url.pathname}${context.url.search}`, {
+          attributes: {
+            [SemanticAttributes.HTTP_HOST]: context.host,
+            [SemanticAttributes.HTTP_URL]: context.url.href,
+            [SemanticAttributes.NET_PEER_IP]: context.remote,
```

Comment:
I don't believe that this is the correct property.

Reply:
Do you know what it should be? I was cross-referencing the official docs and this seemed like the closest one. Happy to look again but I really did do my due diligence here.,

Reply:
No worries Josh I wasn't trying to imply that!! The docs are not good.
If I understand what is contained in `context.remote` correctly, then I think this property should be mapped to `SemanticAttributes.HTTP_CLIENT_IP` (guessing at the right name here)
See this (and the referenced note [2]): https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions

![image](https://user-images.githubusercontent.com/1183853/150859343-80628be4-3bd1-421c-b174-e7846a9c29e1.png)

Reply:
Word! This makes more sense to me than a peer IP. I'll make the change. Thanks for spotting me!